test/CMakeLists.txt

```diff
@@ -179,16 +185,32 @@ macro(try_add_performance_test)
     ${rmw_implementation}
     "ROS2"
     ${rmw_implementation}
+    "async"
   )
   if(COMM_TYPE_${rmw_implementation})
     add_performance_test(
       ${COMM_TYPE_${rmw_implementation}}
       ${COMM_TYPE_${rmw_implementation}}
       ${rmw_implementation}
+      "async"
     )
   else()
     message(STATUS "No standalone DDS support for RMW: ${rmw_implementation}")
   endif()
 endmacro()
 
+# add_performance_test(
+#   "rmw_fastrtps_sync_cpp"
+#   "ROS2"
+#   "rmw_fastrtps_cpp"
+#   "sync"
+# )
```

Comment:
There is no way to run rmw_fastrtps_cpp async with ROS2. Removed

Reply:
Hi all, thanks for all the work you're putting into this!

I'd like to point out that it is possible to run `rmw_fastrtps_cpp` from ROS 2 both in sync and async. As explained in [rmw_fastrtps repo](https://github.com/ros2/rmw_fastrtps), `rmw_fastrtps_cpp` sets (no  matter what the XML configures):

* Publication mode: ASYNCHRONOUS_PUBLISH_MODE. This is done because up to Fast-RPTS v1.9.3 (ros2-eloquent), it is not possible to send fragments synchronously. v1.9.4 is already able to do that, so in upcoming ROS 2 realeases this will not be hard set.
* History memory policy: PREALLOCATED_WITH_REALLOC_MEMORY_MODE. This is done to prevent failures in unbounded messages types.

However, it is possible to overcome this behaviour by setting environment variable RMW_FASTRTPS_USE_QOS_FROM_XML=1. Then, the xml configuration is applied if exxisting, else, the attributes default to the Fast-RTPS default, which are:

* Publication mode: SYNCHRONOUS_PUBLISH_MODE
* History memory policy: PREALLOCATED_MEMORY_MODE. This has the risk of not being able to manage unbounded message types. However, if all the message types are bounded then it is safe to leave by default.

If you need some help setting this up, I'll be happy to give a hand!